pinot-core/src/main/java/org/apache/pinot/core/common/DataFetcher.java

```diff
@@ -415,10 +421,12 @@ private ForwardIndexReaderContext getReaderContext() {
     }
 
     void readDictIds(int[] docIds, int length, int[] dictIdBuffer) {
+      Tracing.activeRecording().setInputDataType(_dataType, _singleValue);
```

Comment:
Should we check if the recording is enabled before setting the data type? In case some implementations do implement this method, but have `_enabled` set dynamically

Reply:
no, I don't think so.

Reply:
So the contract here is that `InvocationRecording` implementation should not assume the recording methods are not called when it is disabled. Might worth adding it into the javadoc of the interface

Reply:
But `InvocationRecording` isn't in control, the tracer decides what is enabled and what isn't. The only reason for `isEnabled()` in the first place is to allow instrumentation to save a few cycles here and there without needing to rely on everything getting inlined and DCEd.